### ISSUE TITLE
[SOLVED] ANDROID only: Flutter 3.1.0 complaining for exact version of `firebase_messaging:@[19.0.0, 22.0.99]` instead of range?!?!

### ISSUE THREAD

**Issue by jonchui** [OneSignal/OneSignal-Flutter-SDK#451]
**Description:**

**TL;DR - Last I checked, 22.0.0 is between 19.0.0 and 22.0.99 ?!?! LOL**

I'm using 3.1.0 in a flutter app and iOS works fine. But when I build for android, i get the weirdest error, when running `flutter run`  (which on android, is running `./gradlew --info :app:assembleDebug`):

```
> In project 'app' a resolved Google Play services library dependency depends on another at an exact version (e.g. "[19.0.
  0, 22.0.99]", but isn't being resolved to that version. Behavior exhibited by the library will be unknown.

  Dependency failing: com.onesignal:OneSignal:4.4.1 -> com.google.firebase:firebase-messaging@[19.0.0, 22.0.99], but fireb
  ase-messaging version was 22.0.0.
```

I thought the `[` meant a range of versions? Of which i'm pretty sure `22.0.0` is between 19.0.0 and `22.0.99` ?!?! 

Anyways - I'll file a ticket for this too - but just wondering if anyone else was getting this... 
seems that 

From the text error, it seems that android build system is mistakenly thinking that `@[19.0.0, 22.0.99]` is an `exact version` ?

**Environment**

1. Added `onesignal: ` to `pubfile.yaml` (at first no specific version)
2. Tried to install on android via `flutter run` with android device AND simulators

Here's a LOOM:
<a href="https://www.loom.com/share/cc7aadad240e4031b5c9e2ed82662add">
    <p>weird onesignal flutter 3.1.0 running error on android - Watch Video</p>
    <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/cc7aadad240e4031b5c9e2ed82662add-with-play.gif">
  </a>

**Anything else:**

(crash stacktraces, as well as any other information here)

- Tried adding `googleServices { disableVersionCheck = false }` to the `app/build.gradle` file still got the same error

```
Launching lib/main.dart on sdk gphone x86 in debug mode...
lib/main.dart:1

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':onesignal_flutter:generateDebugRFile'.
> In project 'app' a resolved Google Play services library dependency depends on another at an exact version (e.g. "[19.0.
  0, 22.0.99]", but isn't being resolved to that version. Behavior exhibited by the library will be unknown.

  Dependency failing: com.onesignal:OneSignal:4.4.1 -> com.google.firebase:firebase-messaging@[19.0.0, 22.0.99], but fireb
  ase-messaging version was 22.0.0.

  The following dependencies are project dependencies that are direct or have transitive dependencies that lead to the art
  ifact with the issue.
  -- Project 'app' depends on project 'onesignal_flutter' which depends onto com.google.firebase:firebase-messaging@{stric
  tly 22.0.0}
  -- Project 'app' depends on project 'onesignal_flutter' which depends onto com.onesignal:OneSignal@{strictly 4.4.1}
  -- Project 'app' depends on project 'onesignal_flutter' which depends onto com.onesignal:OneSignal@4.4.1

  For extended debugging info execute Gradle from the command line with ./gradlew --info :app:assembleDebug to see the dep
  endency paths to the artifact. This error message came from the google-services Gradle plugin, report issues at https://
  github.com/google/play-services-plugins and disable by adding "googleServices { disableVersionCheck = false }" to your b
  uild.gradle file.

* Try:
```

Pubspec.yaml here:

```
name: firebase_google_sign_in_tutorial
description: A new Flutter project.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev
version: 1.0.0+1

environment:
  sdk: '>=2.12.0 <3.0.0'

dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^1.0.2
  firebase_auth: ^1.2.0
  firebase_core: ^1.2.0
  font_awesome_flutter: ^9.0.0
  google_sign_in: ^5.0.5
  http:
  onesignal_flutter: ^3.0.0
  webview_flutter:
  flutter_chat_ui: ^1.1.9
  file_picker:
  image_picker:
  mime:
  open_file:
  uuid:

dev_dependencies:
  flutter_test:
    sdk: flutter

flutter:
  assets:
    - assets/images/main-img.png
    - assets/images/sign-in.png
```

**Comment by jonchui** [comment]
Okay, I think i found the offending plugin in my `app/build.gradle` file

```
apply plugin: 'com.google.gms.google-services'
```

When I remove it, that error doesn't show up.

**Comment by jonchui** [comment]
Okay for those who can learn from my stupidity, I finally fixed this:
- by actually reading the rest of the directions... (sigh... ) specifically, adding [Step 5](https://documentation.onesignal.com/docs/flutter-sdk-setup#step-5-set-up-onesignal-for-android) from the readme to the build.gradle in android
- changing my example files applicationIds to the ones that matched my `google-services.json` file
- adding a weird dep line to my `app/build.gradle`:
```
// FIXES weird duplicate error source: https://stackoverflow.com/questions/56639529/duplicate-class-com-google-common-util-concurrent-listenablefuture-found-in-modu
    implementation 'com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava'
```

![image](https://user-images.githubusercontent.com/204868/127674655-baa58e07-35d4-4746-b635-befdc1299a2e.png)

**Comment by emawby** [comment]
@jonchui I am glad the issue is resolved. Thank you for the feedback on the documentation as well
